source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+Building directly from dockerhub
+================================
+
+There are multiple ways to build a container. To build directly from docker hub, for example the latest version of tensorflow, one can invoke:
+
+.. code-block:: bash
+   
+   singularity build --nv tf_latest.sif docker://tensorflow/tensorflow:latest
+
+and the image ``tf_latest.sif`` from `dockerhub <https://hub.docker.com>`_ will be built, containing the contents of the latest ``tensorflow`` image from the makers of tensorflow. You can also get an image from a different source, such as the Nvidia container repository:
+
+.. code-block:: bash
+
+   singularity build --nv nvidia-tf.sif docker://nvcr.io/nvidia/tensorflow:22.07-tf2-py3
+
+Such an image will contain all the necessary drivers and compile flags to run on Nvidia GPUs, which is preferable on :abbr:`Spider (Symbiotic Platform(s) for Interoperable Data Extraction and Redistribution)`.
+
+To directly run the container without writing to disk invoke:
+
+.. code-block:: bash
+
+   singularity run --nv docker://nvcr.io/nvidia/tensorflow:22.07-tf2-py3
+
+In the examples below, the base images are taken from the internet and expanded upon using *definition* files, to build custom singularity containers.
+
+Running CUDA code 
+=================
+
+
+Here, we show the method of using a definitions file, as opposed to above, where directly building from a repository is shown. The contents of the definitions file are shown before these contents are explained. Start by making the file called ``cuda_example.def`` and add all the steps we want to take to make a container:
+
+.. code-block:: bash
+   
+   Bootstrap: docker
+   From: nvidia/cuda:11.7.0-devel-centos7
+
+   %post
+   #This section is run inside the container 
+   yum -y install git make
+   mkdir /test_repo
+   cd /test_repo
+   git clone https://github.com/NVIDIA/cuda-samples.git
+   cd /test_repo/cuda-samples/Samples/2_Concepts_and_Techniques/eigenvalues/
+   make
+
+   %runscript
+   #Executes when the "singularity run" command is used
+   #Useful when you want the container to run as an executable
+   cd /test_repo/cuda-samples/Samples/2_Concepts_and_Techniques/eigenvalues/
+   ./eigenvalues
+
+   %help
+   This is a demo container to show how to build and run a CUDA application
+   on a GPU node
+
+This container will take a base image from `docker-hub <https://hub.docker.com/>`_ and use pre-built `nvidia/cuda <https://hub.docker.com/r/nvidia/cuda>`_ container of a specific version. This container also contains the necessary CUDA tools to compile binaries that run on GPUs. After starting from this base-image, in the next steps some tools are installed, directories are created and filled with a git repository. From this repository a single example of a CUDA applictation is compiled. When running the container on the command line, this application is run automatically.
```

Comment:
typo: use a pre-built
why is 'a specific version' necessary for this example, can we not use some latest version?

Reply:
For this container there is only a specific versioned release.